src/v/cloud_roles/signature.cc

```diff
@@ -392,7 +426,10 @@ std::error_code signature_v4::sign_header(
     if (!canonical_req) {
         return canonical_req.error();
     }
-    vlog(clrl_log.trace, "\n[canonical-request]\n{}\n", canonical_req.value());
+    vlog(
```

Comment:
nit: since it's now doing a bit of work, maybe wrap this into `if(is_trace_level_enabled) vlog(....)`  
I remember there was a pr to have a macro for this, but I don't see it in the tree.

Reply:
What is `is_trace_level_enabled`?

Does something like this do the equivalent:

```
    if (clrl_log.is_enabled(seastar::log_level::trace)) {
        vlog(
          clrl_log.trace,
          "\n[canonical-request]\n{}\n",
          redact_headers_from_string(canonical_req.value()));
    }
```

Reply:
yes, i did not have a reference for it